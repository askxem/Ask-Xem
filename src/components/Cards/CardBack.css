.container {
    display: flex;
    flex-direction: column;
    align-items: center;
    justify-content: center;
    padding: 10px;    
    width: 70%;

}

figure {
    padding: 10px;
    margin: 0;
    display: flex;
    flex-direction: column;
    justify-content: center;
    align-items: center;
    border: 1px solid gray;
    border-radius: 20px;
}

.image {
    /* max-width: 40vw; */
    width: 40%;
}

.heart {
    width: 10vw;
}
