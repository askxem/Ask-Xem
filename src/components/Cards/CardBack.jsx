import styles from './CardBack.css'

export default function CardBack({ card }) {
    return (
        <div className={styles.container}>
            <figure>
                <h3>{card.title}</h3>
                <figcaption className={styles.definition}>
                {card.definition}
                </figcaption>
                <a href={card.source}>Source</a>
                <img
                    src={`/icons/${card.animal}.png`}
                    alt={card.animal}
                    className={styles.image}
                />
        </figure>
      </div>
    )
}
