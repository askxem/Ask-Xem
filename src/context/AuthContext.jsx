import { createContext, useState, useContext, useMemo } from "react";
import { getUser, signInUser, signOutUser, signUpUser } from "../services/users.js";


const AuthContext = createContext();

function ProvideAuth({children}) {

    const currentUser = getUser();

    const [user, setUser] = useState(
        currentUser ? {...currentUser} : {}
    );

    async function signUp(email, password) {
        const newUser = await signUpUser(email, password);
        setUser(newUser);
    }

    async function signIn(email, password) {
        const newUser = await signInUser(email, password);
        setUser(newUser);
    }

    async function signOut() {
        await signOutUser();
        localStorage.clear();
        setUser({});
        history.push('/home');
    }

    const value = useMemo(() => ({ user, signUp, signIn, signOut}), [user]);

    return (
        <AuthContext.Provider value={value} >
            {children}
        </AuthContext.Provider>
    )
}

/**
 * 
 * @returns a user object, a signup, signin, and signout fn that hits the database.
 */
function useAuth(){
    const context = useContext(AuthContext);

    if (context === undefined) throw new Error('Auth context not accessible outside of the auth provider.')

    return context;
}


export { ProvideAuth, useAuth }