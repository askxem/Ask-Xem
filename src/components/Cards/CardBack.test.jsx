import { rest } from 'msw'
import { setupServer } from 'msw/node'
import { render, screen } from '@testing-library/react'
import { MemoryRouter } from 'react-router-dom'
import { ProvideAuth } from '../../context/AuthContext.jsx';
import CardBack from './CardBack';

jest.mock('../../services/users.js');
jest.mock('../../context/AuthContext.jsx')

const mockFavIdResponse = [{ 'card_id': 1 }]
const url = process.env.SUPABASE_URL + '/rest/v1'
const favsServer = setupServer(
    rest.get(url + '/favs', (req, res, ctx) => { return res(ctx.json(mockFavIdResponse)) }
    ))

describe('Favorites view', () => {
    beforeAll(() => {
        favsServer.listen()
    })
    afterAll(() => {
        favsServer.close()
    })

    it('should display the back of the card', async () => {
        const { container } = render(
            <ProvideAuth mockUser={{ id: 2 }}>
                <MemoryRouter>
                    <CardBack card={{
                        id: 1,
                        title: 'Janelle Monae',
                        animal: 'human',
                        definition: 'perfection incarnate',
                        image: 'url.png'
                    }} favStatus={true} />
                </MemoryRouter>
            </ProvideAuth>
        )

        expect(container).toMatchSnapshot()
        await screen.findByText('Janelle Monae')

    })
})