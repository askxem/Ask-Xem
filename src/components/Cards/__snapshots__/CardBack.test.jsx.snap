// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Favorites view should display the back of the card 1`] = `
<div>
  <div
    class="container"
  >
    <figure
      class="backcontainer"
      style="opacity: 0; transform: translateX(100vw) translateZ(0);"
    >
      <div
        class="heartcontainer"
      >
        <img
          alt="heart"
          class="heart"
          src="/redheart.png"
        />
      </div>
      <h3>
        Janelle Monae
      </h3>
      <figcaption
        class="definition"
      >
        perfection incarnate
      </figcaption>
      <a>
        Source
      </a>
      <img
        alt="human"
        class="image"
        src="/icons/human.png"
      />
    </figure>
    <button
      class="backbutton"
      style="opacity: 0;"
    >
      Back
    </button>
  </div>
</div>
`;
